pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -180,9 +180,31 @@ func loadMonConfig(clientset kubernetes.Interface, namespace string) (map[string
 }
 
 func createClusterAccessSecret(clientset kubernetes.Interface, namespace string, clusterInfo *cephconfig.ClusterInfo, ownerRef *metav1.OwnerReference) error {
-	logger.Infof("creating mon secrets for a new cluster")
+	logger.Infof("creating csi and mon secrets for a new cluster")
 	var err error
 
+	// Store the admin secret for the csi driver
+	csiSecrets := map[string][]byte{
+		// adminID is expected for the cephfs driver
+		"adminID":  []byte("admin"),
+		"adminKey": []byte(clusterInfo.AdminSecret),
+		// userID is expected for the cephfs driver
+		"userID":  []byte("admin"),
+		"userKey": []byte(clusterInfo.AdminSecret),
+	}
+	csiSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-csi",
+			Namespace: namespace,
+		},
+		Data: csiSecrets,
```

Comment:
can we use `stringData` instead of Data?   we can skip `[]byte` conversion above

Reply:
@netzzer has reported that to work on OpenShift she has seen that `Data` is required, while `StringData` fails. I don't recall the details though.

Reply:
for confirmation:
@nehaberry are you using `stringData` or `Data` in secret in OCP

Reply:
@Madhu-1 yes we are using "StrinData" in the yaml. Once the resource is created, it automatically shows up as Data with base64 encoded value;


e.g.

1. Noted the key for admin

sh-4.2# ceph auth get-key client.admin
AQD1wUdd8KNCOBAA+R58MhsIODdSKL8WakpXCA==sh-4.2# 


**2. Created the secret yaml**

$ cat /tmp/Secretti_vr_p_
apiVersion: v1
kind: Secret
metadata:
  name: secret-test-rbd-0512275256
  namespace: openshift-storage
**stringData:
  userID: admin
  userKey: AQD1wUdd8KNCOBAA+R58MhsIODdSKL8WakpXCA==**

-----------
**3. Created the secret :** 

oc -n openshift-storage --kubeconfig /home/nberry/aws-install/aug5-1/auth/kubeconfig create -f /tmp/Secretti_vr_p_ -o yaml

**5. Verified the data in created secret**
$ oc get secret secret-test-rbd-0512275256 -n openshift-storage -o yaml
apiVersion: v1
**data:
  userID: YWRtaW4=
  userKey: QVFEMXdVZGQ4S05DT0JBQStSNThNaHNJT0RkU0tMOFdha3BYQ0E9PQ==**
kind: Secret
metadata:
  creationTimestamp: "2019-08-05T06:57:53Z"
  name: secret-test-rbd-0512275256
  namespace: openshift-storage
  resourceVersion: "41528"
  selfLink: /api/v1/namespaces/openshift-storage/secrets/secret-test-rbd-0512275256
  uid: 58ba319a-b74e-11e9-8637-06e8dff050ca
type: Opaque


6. Able to create pvc

$ oc get sc storageclass-test-rbd-0512280619 -o yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: "2019-08-05T06:58:07Z"
  name: storageclass-test-rbd-0512280619
  resourceVersion: "41607"
  selfLink: /apis/storage.k8s.io/v1/storageclasses/storageclass-test-rbd-0512280619
  uid: 60f0ac13-b74e-11e9-b036-0284beed546e
parameters:
  clusterID: openshift-storage
  csi.storage.k8s.io/node-stage-secret-name: secret-test-rbd-0512275256
  csi.storage.k8s.io/node-stage-secret-namespace: openshift-storage
  csi.storage.k8s.io/provisioner-secret-name: secret-test-rbd-0512275256
  csi.storage.k8s.io/provisioner-secret-namespace: openshift-storage
  imageFeatures: layering
  imageFormat: "2"
  pool: cbp-test-0512275940
provisioner: rbd.csi.ceph.com
reclaimPolicy: Delete
volumeBindingMode: Immediate

Reply:
When using the client-go in rook, the api will do the base64 coding automatically so it should be the same either way for rook to use data or stringData. I'll stick with data since that's what I tested and is working.